### ISSUE TITLE
[실험] 지식 증류 학습 구현 및 실행 ⭐

### ISSUE THREAD

**Issue by YOONPYOGitHub** [YOONPYOGitHub/knowledge-distillation#24]
## 목표
프로젝트 핵심 — KD 학습 파이프라인 구현 및 실행

## 세부 작업
- [ ] KD Loss 함수 구현 (KL Divergence + Temperature)
- [ ] Total Loss: `α·CE + (1-α)·T²·KL`
- [ ] Training loop (Teacher frozen → Student 학습)
- [ ] Validation loop + Checkpoint 저장
- [ ] 소규모 테스트 (1 epoch, 100 step)
- [ ] 전체 학습 실행 (3~5 epochs)

## 핵심 수식
```
KD Loss = α · CE(Student, Label) + (1-α) · T² · KL(Student ∥ Teacher)
```

## 동시에 진행
- [ ] 베이스라인 학습 (CE Loss만, 동일 조건)
- [ ] 학습 모니터링 (loss 곡선, GPU 메모리)

## 참고
- 기존 이슈 #8, #9, #10 내용 통합
- `README.md` — [핵심 알고리즘 상세](#6-핵심-알고리즘-상세), 수도코드

> 💡 바이브 코딩으로 구현 후, 학습 과정 관찰 및 loss 수렴 확인에 집중

## 완료 기준
- [ ] KD 학습 + FT 학습 모두 완료, loss 감소 확인

**Comment by YOONPYOGitHub** [comment]
새로운 1년 학습 로드맵 기반 이슈로 재설계합니다.
